tools/reactive-controllers/src/MatchMedia.ts

```diff
@@ -15,35 +15,153 @@ export const DARK_MODE = '(prefers-color-scheme: dark)';
+        this.onChange();
+
         host.addController(this);
     }
 
+    /**
+     * Determines the determined screen size based on which media query list object matches the current screen width.
+     *
+     * @returns the determined screen size value
+     */
+    private _determineScreenSize(): 'small' | 'large' {
+        if (this.smallScreenQuery.matches) {
+            return 'small';
+        }
+        return 'large';
+    }
+
+    /**
+     * Determines the determined screen orientation based on which media query list object matches the current screen width.
+     *
+     * @returns the determined screen orientation value
+     */
+    private _determineScreenOrientation(): 'portrait' | 'landscape' {
+        if (this.orientationQuery.matches) {
+            return 'portrait';
+        }
+        return 'landscape';
+    }
+
     public hostConnected(): void {
-        this.media?.addEventListener('change', this.onChange);
+        // to support the case where the query is provided
+        this.mediaQueryList?.addEventListener(
+            'change',
+            this.onChange.bind(this)
+        );
+
+        this.isMobileScreenQuery?.addEventListener(
+            'change',
+            this.onChange.bind(this)
+        );
+        this.smallScreenQuery?.addEventListener(
+            'change',
+            this.onChange.bind(this)
+        );
+        this.largeScreenQuery?.addEventListener(
+            'change',
+            this.onChange.bind(this)
+        );
+        this.orientationQuery?.addEventListener(
+            'change',
+            this.onChange.bind(this)
+        );
     }
 
     public hostDisconnected(): void {
-        this.media?.removeEventListener('change', this.onChange);
+        // to support the case where the query is provided
+        this.mediaQueryList?.removeEventListener(
+            'change',
+            this.onChange.bind(this)
+        );
+
+        this.isMobileScreenQuery?.removeEventListener(
+            'change',
+            this.onChange.bind(this)
+        );
+        this.smallScreenQuery?.removeEventListener(
+            'change',
+            this.onChange.bind(this)
+        );
+        this.largeScreenQuery?.removeEventListener(
+            'change',
+            this.onChange.bind(this)
+        );
+        this.orientationQuery?.removeEventListener(
+            'change',
+            this.onChange.bind(this)
+        );
     }
 
-    protected onChange(event: MediaQueryListEvent): void {
-        if (this.matches === event.matches) return;
-        this.matches = event.matches;
-        this.host.requestUpdate(this.key, !this.matches);
+    protected onChange(): void {
+        // if the query is provided
+        if (this.mediaQueryList) {
+            this.matches = this.mediaQueryList.matches;
+            this.host.requestUpdate();
+            return;
+        }
+
+        const screenSize = this._determineScreenSize();
```

Comment:
Could we encapsulate this "check if is mobile" logic into a dedicated method?

```ts
private _checkIsMobile(): boolean {
  const screenSize = this._determineScreenSize();
  const orientation = this._determineScreenOrientation();
  const isMobileScreen = this.isMobileScreenQuery.matches;

  // if the screen size is small or the orientation is landscape and the device is mobile
  return isMobileScreen && (screenSize === screenSize.small || orientation === orientation.landscape);
}
```

and then

```ts
this.matches = _checkIsMobile()
```